models/DataObject/Traits/DefaultValueTrait.php

```diff
@@ -130,7 +117,27 @@ protected function handleDefaultValue($data, $object = null, $params = [])
                 }
             }
 
-            $data = $this->doGetDefaultValue($object, $context);
+            // we check first if we even want to work with default values. if this is not the case then
+            // we are also not allowed to inspect the parent value.
+
+            // if the parent doesn't have a value then we take the configured value as fallback
+            $configuredDefaultValue = $this->doGetDefaultValue($object, $context);
+            if (!$this->isEmpty($configuredDefaultValue)) {
+
+                if ($class && $class->getAllowInherit()) {
+                    $params = [];
+
+                    try {
+                        $data = $owner->getValueFromParent($this->getName(), $params);
+                        if (!$this->isEmpty($data)) {
+                            return $data;
```

Comment:
if parent has value then it should return null so as to inherit the value form parent. right?

Reply:
No. It should return the value from the parent as this will go directly into the resource

Reply:
please add a note in docs, if not there :)